### ISSUE TITLE
Multiclass MLP and weights

### ISSUE THREAD

**Issue by rretureau** [dnouri/nolearn#187]
Hi,

I managed to run a binary network using nolearn (really nice and easy, thanks a lot). I need to do the same work on three class. I use the following code but my problem is that it predict all in the first class. I first assumed it was a weight problem of training datas and managed to uniform weight of class. It doesn't change anything, the accuracy is 0.33 (all in the first class and 0 predicted in the other). 

Here is the code
```python
net1 = NeuralNet(
    layers=[  # three layers: one hidden layer
        ('input', layers.InputLayer),
        ('hidden1', layers.DenseLayer),
        ('output', layers.DenseLayer),
        ],
    # input layer
    input_shape=(None, 220),  # 11x20 input pssm profile per batch
    # hidden layers
    hidden1_num_units=100,
    hidden1_nonlinearity=lasagne.nonlinearities.sigmoid,
    # output layer
    output_nonlinearity=lasagne.nonlinearities.softmax,
    output_num_units=3,
    # optimization method:
    update=nesterov_momentum,
    update_learning_rate=0.01,
    update_momentum=0.9,
    regression=False,  # flag to indicate we're dealing with regression problem
    max_epochs=50,  # we want to train this many epochs
    verbose=1,
    )

if __name__ == "__main__":
    X,y = load(sys.argv[1])
    X_t, y_t = load(sys.argv[2])

    nn1 = net1.fit(X,y)
    pred1 = net1.predict(X_t)

    cfm1 = confusion_matrix(y_t, pred1)
    print cfm1 
```

I can't see where the problem came from. As you can see, I have a single Array with classes and the corresponding attributes (all are well formated and works well in binary system). 220 input and 3 output. Do you see anything that can explain my problem ? 

My other question is about weight. Is it important that all the class have the same representation in the training datas ? Or is there a way to specify the network these weights ?

**Comment by rretureau** [comment]
Ok, I read the other issues and saw that weight are not supported in the current version. Also, I figured out where the problem come from. Actually, the expected input (y) for classification has to start from 0 (mine started from 1). 

Thanks ;)

**Comment by dnouri** [comment]
> Ok, I read the other issues and saw that weight are not supported in the current version.

Felix has [one way to do weighting](https://github.com/dnouri/nolearn/issues/109#issuecomment-196364054):

```python
def custom_objective(...):
    ...
    weights_per_label = theano.shared(lasagne.utils.floatX([0.8, 0.1, 0.1]))
    weights = weights_per_label[target]  # This is a bit non-obvious
    loss = aggregate(loss_function(network_output, target), weights=weights)
    ...
```
